apps/speed.c

```diff
@@ -106,6 +108,19 @@
 # endif
 #endif
 
+#if defined(OPENSSL_SYS_UNIX) && defined(OPENSSL_THREADS)
+# include <unistd.h>
+# if _POSIX_VERSION >= 200112L
+#  define ASYNC_POSIX
+# endif
+#elif defined(_WIN32) || defined(__CYGWIN__)
+# define ASYNC_WIN
+#endif
+
+#if !defined(ASYNC_POSIX) && !defined(ASYNC_WIN)
+# define ASYNC_NULL
+#endif
+
```

Comment:
I am happy to change this part and maybe replace it with a new API that tells whether we are on a platform that is async capable